toolchain/check/declaration_name_stack.cpp

```diff
@@ -39,7 +40,12 @@ auto DeclarationNameStack::Pop() -> NameContext {
     ApplyNameQualifier(parse_node, name_id);
   }
 
-  return declaration_name_stack_.pop_back_val();
+  return declaration_name_stack_.back();
```

Comment:
Done. Turned out to be slightly awkward because we want the original state in the name that we return here, for later use by the caller, but I think this adds some good error checking, thanks!